server/cloudformation/cloudformation.go

```diff
@@ -189,11 +195,17 @@ type CustomResourceProvisioner struct {
 
 // NewCustomResourceProvisioner returns a new CustomResourceProvisioner with an
 // sqs client configured from config.
-func NewCustomResourceProvisioner(db *sql.DB, config client.ConfigProvider) *CustomResourceProvisioner {
+func NewCustomResourceProvisioner(empire *empire.Empire, config client.ConfigProvider) *CustomResourceProvisioner {
 	return &CustomResourceProvisioner{
 		Provisioners: map[string]Provisioner{
 			"Custom::InstancePort": &InstancePortsProvisioner{
-				ports: lb.NewDBPortAllocator(db),
+				ports: lb.NewDBPortAllocator(empire.DB.DB.DB()),
+			},
+			"Custom::EmpireApp": &AppResource{
+				empire: empire,
+			},
+			"Custom:EmpireAppEnvironment": &EnvironmentResource{
```

Comment:
ok, i also renamed the file to `app_environment.go` to handle the conflict with the file name too.